### ISSUE TITLE
Support for retina tiles ? 

### ISSUE THREAD

**Issue by petzlux** [mapfish/mapfish-print#478]
Hi everyone,

Trying to implement high dpi WMTS background layer in our project 
https://github.com/Geoportail-Luxembourg/geoportailv3/pull/1559

Works great in ol3, but the print makes problems. The print result from this pull request is:

![map_geoportal_lu](https://cloud.githubusercontent.com/assets/2540428/20302515/e76c6b12-ab27-11e6-8568-9f5111010fee.png)

I have tried to adjust the matrices tilesize setting to `[512, 512]` but then the background layer goes pink in the print. 

I have seen the following discussion #449 but not sure how this applies here.

**Comment by tsauerwein** [comment]
Yes, this is not yet supported properly (see also https://github.com/mapfish/mapfish-print/issues/154). But it would be an interesting feature to add.

**Comment by sbrunner** [comment]
What should we do in the print to support HiDPI tiles?
As I understand to have HiDPI tiles we should:
* Generate a specific tile set for HiDPI
* Update the sended spec.json to use it

But there is nothing in the print... Did I miss something? Or do we want something better?

**Comment by petzlux** [comment]
we have the tiles ready. we just need the print component to be able to use them ...

**Comment by sbrunner** [comment]
Then what's missing?

**Comment by petzlux** [comment]
can you explain to me the correct spec.json to send to mapfish-print for a 512*512 WMTS layer ? As I said, setting `matrice.tileSize = [512, 512];` does not work as seen in the screenshot above .

**Comment by sbrunner** [comment]
I change the title because there nothing related to HiDPI but it seem related to the 512x512 tiles

**Comment by tsauerwein** [comment]
If I get @petzlux right, they really do have Retina/HighDPI tiles. That is they have a 256x256px tiling scheme, but they are using 512x512px tiles. This is different from using 512x512px tiles for a 512x512px tiling scheme.

**Comment by sbrunner** [comment]
@tsauerwein I don't understand what you say...

**Comment by petzlux** [comment]
@sbrunner I think there is a misunderstanding here. 

We have wmts layers now that have double resolution. ie a 512*512 tile is used for a 256*256 display in ol3 ``tilePixelRatio: 2`  in the layer def. 
https://github.com/Geoportail-Luxembourg/geoportailv3/pull/1559/files#diff-c1b2383c6e5019360cf3824e619d7485R91

Also see http://openlayers.org/en/latest/examples/wmts-hidpi.html for an example of what we achieve. 

Now we want to use the same effect in our prints, ie use the same 512*512 tile to effectively double the dpi of the background layer.

**Comment by sbrunner** [comment]
Actually if we don't have HiDPI tiles the tiles are printed as print resolution (as HiDPI) than we don't have to do something for HiDPI tiles we eventually can do something for non HiDPI tiles...
